dissect/target/loaders/containerimage.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import json
+import logging
+import tarfile
+
+from dissect.target.filesystem import LayerFilesystem
+from dissect.target.filesystems.tar import TarFilesystem
+from dissect.target.loaders.tar import TarSubLoader
+from dissect.target.target import Target
+
+log = logging.getLogger(__name__)
+
+DOCKER_ARCHIVE_IMAGE = {
+    "manifest.json",
+    "repositories",
+}
+
+OCI_IMAGE = {
+    "manifest.json",
+    "repositories",
+    "blobs",
+    "oci-layout",
+    "index.json",
+}
+
+
+class ContainerImageTarSubLoader(TarSubLoader):
+    """Load saved container images.
+
+    Supports both the Docker and OCI image specifications.
+
+    References:
+        - https://snyk.io/blog/container-image-formats/
+        - https://github.com/moby/docker-image-spec/
+        - https://github.com/opencontainers/image-spec/
+    """
+
+    def __init__(self, tar: tarfile.TarFile, *args, **kwargs):
+        super().__init__(tar, *args, **kwargs)
+
+        self.tarfs = None
+        self.manifest = None
+        self.name = None
+        self.config = None
+
+        try:
+            self.tarfs = TarFilesystem(None, _tarfile=tar)
+        except Exception as e:
+            raise ValueError(f"Unable to open {str(tar)} as TarFilesystem: {str(e)}")
+        try:
+            self.manifest = json.loads(self.tarfs.path("/manifest.json").read_text())[0]
+            self.name = self.manifest.get("RepoTags", [None])[0]
+        except Exception as e:
+            raise ValueError(f"Unable to read manifest.json inside docker image filesystem: {str(e)}")
+
+        try:
+            self.config = json.loads(self.tarfs.path(self.manifest.get("Config")).read_text())
+        except Exception as e:
+            raise ValueError(f"Unable to read config inside docker image filesystem: {str(e)}")
+
+    @staticmethod
+    def detect(tar: tarfile.TarFile) -> bool:
+        # NOTE: tar.getmembers() / tar.getnames() is still inefficient for large tarballs, how should be prevent this?
```

Comment:
At least `getmembers()` is cached, so it'll get reused by the `GenericTarSubLoader`. Maybe it's a bit more efficient to do a loop over `getmembers()` and break out once you have seen enough filenames?